bubuku/features/rebalance/change.py

```diff
@@ -0,0 +1,248 @@
+import logging
+
+from bubuku.features.rebalance import BaseRebalanceChange
+from bubuku.features.rebalance.broker import BrokerDescription
+from bubuku.zookeeper import BukuExhibitor
+
+_LOG = logging.getLogger('bubuku.features.rebalance')
+
+
+def distribute(amount, items, weight_key, apply_func):
+    if not items:
+        return
+    items = sorted(items, key=weight_key, reverse=True)
+    ceil_amount = int(amount / len(items))
+    amounts = [ceil_amount for _ in range(0, len(items))]
+    for idx in range(0, min(amount - sum(amounts), len(items))):
+        amounts[idx] += 1
+    for idx in range(0, len(items)):
```

Comment:
I would suggest to remove this `apply_func` if possible. Also, I would suggest to always create new collections and return them. I would not optimize at this level. I think that clarity, which helps understand correctness, is more valuable.